src/main/java/org/springframework/integration/aws/inbound/SqsMessageDrivenChannelAdapter.java

```diff
@@ -88,6 +89,10 @@ public void setResourceIdResolver(ResourceIdResolver resourceIdResolver) {
 		this.simpleMessageListenerContainerFactory.setResourceIdResolver(resourceIdResolver);
 	}
 
+	public void setAutoStartUp(boolean autoStartUp) {
+		super.setAutoStartup(autoStartUp);
```

Comment:
This method is bad.
You have to *override* `setAutoStartup()` and yes call `super.setAutoStartup()` and add `this.simpleMessageListenerContainerFactory.setAutoStartup(autoStartUp);`.
Only after that the test will pass.

`SqsMessageDrivenChannelAdapter` extends `AbstractEndpoint` which has `setAutoStartup()`. Since that one isn't `final` we can simple override it and implement our own logic. But don't break anything we have to invoke `super` anyway.

M-m-m. I don't want to offend you, but let me know if you are familiar with something like *polymorphism* in OOP ?